src/MassTransit.Abstractions/Internals/Extensions/TaskExtensions.cs

```diff
@@ -105,17 +105,32 @@ async Task<T> WaitAsync()
 
             async Task WaitAsync()
             {
-                var delayTask = Task.Delay(Debugger.IsAttached ? Timeout.InfiniteTimeSpan : timeout, cancellationToken);
-
-                var completed = await Task.WhenAny(task, delayTask).ConfigureAwait(false);
-                if (completed == delayTask)
+                var cts = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken);
+                try
                 {
-                    task.IgnoreUnobservedExceptions();
+                    Task delayTask = Task.Delay(Debugger.IsAttached ? Timeout.InfiniteTimeSpan : timeout, cts.Token)
+                        .ContinueWith(t => t, CancellationToken.None, TaskContinuationOptions.ExecuteSynchronously, TaskScheduler.Default);
```

Comment:
This continuation isn't needed, or what value does this add?

Reply:
this is done to avoid throwing cancellation exception later on